### PR TITLE
Feature/automation log filter by license

### PR DESCRIPTION
**Pull request by deanhannigan**
## Description
With the licensing feature enabled, the date range filtering options for an applications automation history will reflect the license of the current tenant. If the license is 'free' the option for 'Past Week' is excluded.
This also adds a general `isOptionEnabled` parameter to the Select component.

Addresses: 
- https://github.com/Budibase/budibase/issues/6675

## Screenshots
When not on the Free license, all options appear in the Date Range picker.

Screenshot 2022-09-28 at 12 17 38

When on the free license, the `Past Week` option is excluded and messaging is added to the table header instructing the user how to go about enabling it.

Screenshot 2022-09-28 at 12 23 58

### CODE REVIEW

File: packages/builder/src/components/portal/overview/automation/HistoryTab.svelte [hunk 1]
```diff
@@ -131,10 +136,19 @@
...
       <div class="select">
         <Select
-          placeholder="Past 30 days"
+          placeholder="All"
           label="Date range"
           bind:value={timeRange}
           options={timeOptions}
+          isOptionEnabled={x => {
+            if (
+              isEnabled(TENANT_FEATURE_FLAGS.LICENSING) &&
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:145]: I think we're okay to stop using this flag. There's no going back now

File: packages/builder/src/components/portal/overview/automation/HistoryTab.svelte [hunk 2]
```diff
@@ -145,6 +159,18 @@
           options={statusOptions}
         />
       </div>
+      {#if isEnabled(TENANT_FEATURE_FLAGS.LICENSING) && licensePlan?.type === "free"}
+        <div class="pro-upgrade">
+          <div class="pro-copy">Store up to 30 days of automations</div>
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:164]: This message could probably be updated, we have more options now than just 30 days: 

Screenshot 2022-09-28 at 13 17 20

Probably the same for the filters, they can go up to 90 days in cloud or unlimited on enterprise.

File: packages/builder/src/components/portal/overview/automation/HistoryTab.svelte [hunk 3]
```diff
@@ -145,6 +159,18 @@
...
+          <Button
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:165]: Similar to the other checks, in cloud we should only enable the upgrade button to account holders. In self host we can leave it on always. For the groups upgrade button I did this: https://github.com/Budibase/budibase/blob/master/packages/builder/src/pages/builder/portal/manage/groups/index.svelte#L125

File: packages/builder/src/components/portal/overview/automation/HistoryTab.svelte [hunk 4]
```diff
@@ -159,14 +161,10 @@
...
-      {#if isEnabled(TENANT_FEATURE_FLAGS.LICENSING) && licensePlan?.type === "free"}
+      {#if (licensePlan?.type !== Constants.PlanType.ENTERPRISE && $auth.user.accountPortalAccess) || !$admin.cloud}
         <div class="pro-upgrade">
-          <div class="pro-copy">Store up to 30 days of automations</div>
-          <Button
-            primary
-            newStyles
-            on:click={window.open(upgradeUrl, "_blank")}
-          >
+          <div class="pro-copy">Expand your automation log history</div>
```
Comment (Rory-Powell) [inline packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:166]: 👍

File: packages/builder/src/components/portal/overview/automation/HistoryTab.svelte [hunk 5]
```diff
@@ -141,11 +142,12 @@
...
-              isEnabled(TENANT_FEATURE_FLAGS.LICENSING) &&
-              licensePlan?.type === "free"
-            ) {
-              return "1-w" !== x.value
+            if (licensePlan?.type === Constants.PlanType.FREE) {
+              return ["1-w", "30-d", "90-d"].indexOf(x.value) < 0
+            } else if (licensePlan?.type === Constants.PlanType.TEAM) {
+              return ["90-d"].indexOf(x.value) < 0
+            } else if (licensePlan?.type === Constants.PlanType.PRO) {
+              return ["30-d", "90-d"].indexOf(x.value) < 0
```
[the remark this answers is not in the document]
Comment (deanhannigan) [reply packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:150]: Hey Rory, for the plans mentioned above, it's disabling the listed options. Business and Enterprise are covered by the final `return true` on 152 as they show all options

[the remark this answers is not in the document]
Comment (Rory-Powell) [reply packages/builder/src/components/portal/overview/automation/HistoryTab.svelte:150]: All good then 👍 Thanks!

**Pull request merged**